src/Features/Core/Portable/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer.cs

```diff
@@ -464,16 +497,24 @@ public override void RemoveDocument(DocumentId documentId)
             {
                 _memberRangeMap.Remove(documentId);
 
-                foreach (var stateSet in _stateManager.GetStateSets(documentId.ProjectId))
+                var stateSets = _stateManager.GetStateSets(documentId.ProjectId);
+
+                foreach (var stateSet in stateSets)
                 {
                     stateSet.Remove(documentId);
+                }
 
-                    var solutionArgs = new SolutionArgument(null, documentId.ProjectId, documentId);
-                    for (var stateType = 0; stateType < s_stateTypeCount; stateType++)
+                Owner?.RaiseBulkDiagnosticsUpdated(raiseEvents =>
```

Comment:
yep, I should have audit all code. I saw some place that checks Owner being null so assumed there probably a test that set it null. just audited the code, all gives in owner. removed null check.